OutOfSchool/OutOfSchool.WebApi/Controllers/ProviderController.cs

```diff
@@ -0,0 +1,145 @@
+        }
+
+        /// <summary>
+        /// Get provider by it's key.
+        /// </summary>
+        /// <param name="id">The key in the database.</param>
+        /// <returns>Provider element with some id.</returns>
+        [HttpGet("{id}")]
+        [ProducesResponseType(StatusCodes.Status200OK)]
+        [ProducesResponseType(StatusCodes.Status400BadRequest)]
+        [ProducesResponseType(StatusCodes.Status401Unauthorized)]
+        public async Task<ActionResult<ProviderDto>> GetProviderById(long id)
+        {
+            if (id == 0)
+            {
+                return BadRequest("Id cannot be 0.");
+            }
+
+            return Ok(await providerService.GetById(id).ConfigureAwait(false));
+        }
+
+        /// <summary>
+        /// Create new provider.
+        /// </summary>
+        /// <param name="providerDTO">Element which must be added.</param>
+        /// <returns>A <see cref="Task{TResult}"/> representing the result of the asynchronous operation.</returns>
+        [Authorize(Roles = "provider,admin")]
+        [HttpPost]
+        [ProducesResponseType(StatusCodes.Status400BadRequest)]
+        [ProducesResponseType(StatusCodes.Status201Created)]
+        [ProducesResponseType(StatusCodes.Status401Unauthorized)]
+        public async Task<ActionResult<Provider>> Create(ProviderDto providerDTO)
+        {
+            if (!ModelState.IsValid)
+            {
+                return BadRequest(ModelState);
+            }
+
+            if (providerDTO == null)
+            {
+                return BadRequest(providerDTO);
+            }
+
+            try
+            {
+                providerDTO.UserId = User.FindFirst("sub")?.Value;
+                ProviderDto provider = await providerService.Create(providerDTO).ConfigureAwait(false);
+                return CreatedAtAction(
+                    nameof(GetProviderById),
+                    new
+                    {
+                        id = provider.Id,
+                    }, provider);
+            }
+            catch (ArgumentException ex)
+            {
+                return BadRequest(ex.Message);
+            }
+        }
+
+        /// <summary>
+        /// Update info about some provider in database.
+        /// </summary>
+        /// <param name="providerDTO">Entity.</param>
+        /// <returns>Provider's key.</returns>
+        [Authorize(Roles = "provider,admin")]
+        [HttpPut]
+        [ProducesResponseType(StatusCodes.Status400BadRequest)]
+        [ProducesResponseType(StatusCodes.Status401Unauthorized)]
+        [ProducesResponseType(StatusCodes.Status200OK)]
+        public async Task<ActionResult> Update(ProviderDto providerDTO)
+        {
+            if (!ModelState.IsValid)
+            {
+                return BadRequest(ModelState);
+            }
+
+            return Ok(await providerService.Update(providerDTO).ConfigureAwait(false));
```

Comment:
please return CreatedAtAction as Mykhailo suggested

Reply:
He said to use CreatedAtAction with Post method, here we have the Put method